software/contrib/bernoulli_gates.py

```diff
@@ -0,0 +1,218 @@
+from europi import *
+from time import sleep_ms
+from random import random
+from europi_script import EuroPiScript
+import machine
+
+# Constant values for display
+BERNOULLI_BAR_LEN = 50
+BERNOULLI_BAR_WID = 6
+BERNOULLI_INDI_WID = 5
+
+# Trigger Time (some module may not be able to catch up the trigger if it is too short)
+BERNOULLI_TRG_T = 20
+
+class SingleBernoulliGate():
+    def __init__(self, control_knob =  k1, control_port = ain, out_port = (cv1, cv2, cv3), visualization_para = ('P1', (0, 0), 0, -3), port3_func = 'clock', port3_source_cv = None):
+        '''
+        1. control_knob and control_port tunes the probability threshold
+        if control_port == None, probability threshold is only determined by control_knob
+        2. out_port assigns left, right, and function port accordingly
+        out_port shoud have at least 2 ports (port3_func == 'none')
+        3. visualization_para = (text, text_pos, tex1_pos, bar_pos_offset)
+        4. port3_func can be 'none', 'clock', 'and', 'or', 'xor'
+        the output of function_port = logic(current left, source), logic can be and, or, xor
```

Comment:
I like the way you made the `port3` stuff flexible here. It's much more elegant than the method that I was thinking of suggesting.

Could you add `port3_source_cv` to the docstring here?